refl1d/reflmodule.py

```diff
@@ -0,0 +1,11 @@
+from .lib_numba.reflectivity import reflectivity_amplitude as _reflectivity_amplitude
+from .lib_numba.magnetic import magnetic_amplitude as _magnetic_amplitude
+from .lib_numba.magnetic import calculate_u1_u3 as _calculate_u1_u3
+from .lib_numba.convolve import convolve_gaussian
+from .lib_numba.convolve import convolve_uniform
+from .lib_numba.convolve_sampled import convolve_sampled
+from .lib_numba.contract_profile import align_magnetic as _align_magnetic
+from .lib_numba.contract_profile import contract_by_area as _contract_by_area
+from .lib_numba.contract_profile import contract_mag as _contract_mag
+from .lib_numba.rebin import rebin_counts
+from .lib_numba.rebin import rebin_counts_2D
```

Comment:
I'm not sure we need a "lib_numba" subdirectory. Assuming we keep it, maybe we should to push more things into the calculation library in case other projects want to use them. Otherwise it feels a bit like an historical accident rather than deliberate design. Regardless, we should probably drop the C library (maybe keeping it around somewhere for a future cuda version) and move lib_numba to lib. Not required for this PR. Put a code cleanup ticket if you think it needs it.


Tests were failing because of a name conflict: reflmodule.py and reflmodule.*.so were competing because the .so files were still in the directory. This will cause problems as we switch between branches and go back in time to check differences in results. I suggest renaming `reflmodule.py` to `refllib.py` or `_reflectivity.py` and updating `reflectivity.py` and `profile.py` accordingly.

Reply:
I moved to the lib_numba layout so that one could switch between the versions easily - they I broke that by naming the aggregator "reflmodule.py"  There are also a couple places where we (collectively) changed the signatures/interfaces of the functions and you can no longer easily switch back and forth.  I don't know how high a priority it is to make it possible to switch back.

The tests are only failing for you because you've previously built the C version - on a clean build (in the CI scripts) the tests are all passing.

Reply:
Yes, on a clean build the tests pass. I'm trying to avoid confusion later when we switch back to an ancient version to do some sort of test or check then forget that we have to remove the .so file when moving back to the current version.

Reply:
It seems that python resolves .so files before .py files?  If it were the other way around that would be more convenient in this case, though maybe not in general.

Reply:
moved to refllib.py